src/Facebook/Authentication/AccessToken.php

```diff
@@ -0,0 +1,223 @@
+ * @package Facebook
+ */
+class AccessToken implements \Serializable
+{
+
+  /**
+   * The access token.
+   *
+   * @var string
+   */
+  protected $accessToken = '';
+
+  /**
+   * A unique ID to identify a client.
+   *
+   * @var string|null
+   */
+  protected $machineId;
+
+  /**
+   * Date when token expires.
+   *
+   * @var \DateTime|null
+   */
+  protected $expiresAt;
+
+  /**
+   * Create a new access token entity.
+   *
+   * @param string $accessToken
+   * @param int $expiresAt
+   * @param string|null machineId
+   */
+  public function __construct($accessToken, $expiresAt = 0, $machineId = null)
+  {
+    $this->accessToken = $accessToken;
+    if ($expiresAt) {
+      $this->setExpiresAtFromTimeStamp($expiresAt);
+    }
+    $this->machineId = $machineId;
+  }
+
+  /**
+   * Generate an app secret proof to sign a request to Graph.
+   *
+   * @param string $appSecret The app secret.
+   *
+   * @return string
+   */
+  public function getAppSecretProof($appSecret)
+  {
+    return hash_hmac('sha256', $this->accessToken, $appSecret);
+  }
+
+  /**
+   * Setter for expires_at.
+   *
+   * @param int $timeStamp
+   */
+  protected function setExpiresAtFromTimeStamp($timeStamp)
+  {
+    $dt = new \DateTime();
+    $dt->setTimestamp($timeStamp);
+    $this->expiresAt = $dt;
+  }
+
+  /**
+   * Getter for expiresAt.
+   *
+   * @return \DateTime|null
+   */
+  public function getExpiresAt()
+  {
+    return $this->expiresAt;
+  }
+
+  /**
+   * Getter for machineId.
+   *
+   * @return string|null
+   */
+  public function getMachineId()
+  {
+    return $this->machineId;
+  }
+
+  /**
+   * Determines whether or not this is an app access token.
+   *
+   * @return bool
+   */
+  public function isAppAccessToken()
+  {
+    return strpos($this->accessToken, '|') !== false;
+  }
+
+  /**
+   * Determines whether or not this is a long-lived token.
+   *
+   * @return bool
+   */
+  public function isLongLived()
+  {
+    if ($this->isAppAccessToken()) {
+      return true;
+    }
+
+    if ($this->expiresAt) {
+      return $this->expiresAt->getTimestamp() > time() + (60 * 60 * 2);
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks the expiration of the access token.
+   *
+   * @return boolean|null
+   */
+  public function isExpired()
+  {
+    if ($this->isAppAccessToken()) {
+      return false;
+    }
+
+    if ($this->getExpiresAt() instanceof \DateTime) {
+      return $this->getExpiresAt()->getTimestamp() < time();
+    }
+
+    return null;
+  }
+
+  /**
+   * Ensures the machine ID from the access token is what we expect.
+   *
+   * @param string $machineId
+   *
+   * @throws FacebookSDKException
+   */
+  public function validateMachineId($machineId)
+  {
+    if ($this->machineId !== $machineId) {
+      throw new FacebookSDKException('Access token contains unexpected machine ID.', 401);
+    }
+  }
+
+  /**
+   * Returns the access token as a string.
+   *
+   * @return string
+   */
+  public function asString()
```

Comment:
As you renamed the attribute to `$value`, I think `getValue()` makes more sense. `asString()` sementicaly duplicate the native `__toString()`